mm2src/coins/watcher_common.rs

```diff
@@ -0,0 +1,37 @@
+use crate::{ValidatePaymentError, INVALID_WATCHER_REWARD_ERR_LOG};
+use mm2_err_handle::prelude::MmError;
+
+pub const REWARD_GAS_AMOUNT: u64 = 70000;
+const REWARD_MARGIN: f64 = 0.05;
+
+pub fn validate_watcher_reward(
+    expected_reward: u64,
+    actual_reward: u64,
+    is_exact: bool,
+) -> Result<(), MmError<ValidatePaymentError>> {
+    if is_exact {
+        if actual_reward != expected_reward {
+            return MmError::err(ValidatePaymentError::WrongPaymentTx(format!(
+                "Payment tx reward_amount arg {} is invalid, expected {}",
+                actual_reward, expected_reward,
+            )));
+        }
+        Ok(())
+    } else {
+        let min_acceptable_reward = get_reward_lower_boundary(expected_reward);
+        let max_acceptable_reward = get_reward_upper_boundary(expected_reward);
+        if actual_reward < min_acceptable_reward || actual_reward > max_acceptable_reward {
```

Comment:
What are the reasons that can make `actual_reward > max_acceptable_reward`? Just wondering :)

Reply:
I decided that it would be better to not allow randomly high rewards, but this check is not actually necessary for security, if a party decides to put too much reward they are free to do so :)